src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs

```diff
@@ -71,16 +71,134 @@ public void ConsensusManager_Fork_Occurs_MinerNode_Reorgs_AndResyncs_ToBestHeigh
+                var minerB = builder.CreateStratisPosNode(this.network);
+                var syncer = builder.CreateStratisPosNode(this.network);
+
+                builder.StartAll();
+
+                minerA.NotInIBD().WithWallet();
+                minerB.NotInIBD();
+                syncer.NotInIBD();
+
+                // MinerA mines to height 15.
+                TestHelper.MineBlocks(minerA, walletName, walletPassword, walletAccount, 15);
+
+                // Sync the network to height 15.
+                TestHelper.Connect(syncer, minerA);
+                TestHelper.Connect(syncer, minerB);
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(syncer, minerA));
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(syncer, minerB));
+
+                // Disconnect Miner A and B.
+                TestHelper.Disconnect(syncer, minerA);
+                TestHelper.Disconnect(syncer, minerB);
+
+                // Ensure syncer does not have any connections.
+                TestHelper.WaitLoop(() => !TestHelper.IsNodeConnected(syncer));
+
+                // Miner A stakes a coin that increases the network height to 16.
+                var minter = minerA.FullNode.NodeService<IPosMinting>();
+                minter.Stake(new WalletSecret() { WalletName = walletName, WalletPassword = walletPassword });
+
+                TestHelper.WaitLoop(() =>
+                {
+                    return minerA.FullNode.ConsensusManager().Tip.Height == 16;
+                });
+
+                minter.StopStake();
+
+                // Update the network consensus options so that the GetStakeMinConfirmations returns a higher value
+                // to ensure that the InvalidStakeDepth exception can be thrown.
+                minerA.FullNode.Network.Consensus.Options = new ConsensusOptionsTest();
+                minerB.FullNode.Network.Consensus.Options = new ConsensusOptionsTest();
+                syncer.FullNode.Network.Consensus.Options = new ConsensusOptionsTest();
+
+                // Syncer now connects to both miners causing a InvalidStakeDepth exception to be thrown
+                // on Miner A.
+                TestHelper.Connect(syncer, minerA);
+                TestHelper.Connect(syncer, minerB);
+
+                // Ensure that Syncer is synced with MinerB.
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(syncer, minerB));
+
+                // Ensure that Syncer is not connected to MinerA.
+                TestHelper.WaitLoop(() => !TestHelper.IsNodeConnectedTo(syncer, minerA));
+
+                Assert.True(syncer.FullNode.ConsensusManager().Tip.Height == 15);
+                Assert.True(minerB.FullNode.ConsensusManager().Tip.Height == 15);
+            }
+        }
+
+        [Fact]
+        public void ConsensusManager_Fork_Occurs_Node_Gets_Disconnected_Due_To_MaxReorgViolation()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
```

Comment:
Wehre do we set the max reorg violation to 15?

Reply:
It is `500` by default on StratisTest (passed into `Consensus` constructor when the `Network` is instantiated), but is being overridden at the top of this source file to `20`.

Note that there is also a `MaxReorgLength` in `ChainState` that should probably be removed, judging by various TODOs.

Reply:
Yes sorry I meant I think in this test he is overriding it to something smaller, so just asking where this happens.

Reply:
Hey @dangershony, yes I am setting it via reflection in the constructor of the test class 👍